libvips/foreign/magick.c

```diff
@@ -109,37 +100,14 @@ magick_sniff(const unsigned char *bytes, size_t length)
 		memcmp(bytes + 4, "ftyp", 4) != 0)
 		return "TGA";
 
-	guint32 ifd_offset = 0;
-
-	/* Big-endian TIFF header
+	/* Try to search the internal magic list for a match.
 	 */
-	if (length >= 8 &&
-		bytes[0] == 'M' &&
-		bytes[1] == 'M' &&
-		bytes[2] == 0 &&
-		bytes[3] == 42)
-		ifd_offset = ((guint32) bytes[4] << 24) |
-			((guint32) bytes[5] << 16) |
-			((guint32) bytes[6] << 8) |
-			bytes[7];
-
-	/* Little-endian TIFF header
-	 */
-	if (length >= 8 &&
-		bytes[0] == 'I' &&
-		bytes[1] == 'I' &&
-		bytes[2] == 42 &&
-		bytes[3] == 0)
-		ifd_offset = ((guint32) bytes[7] << 24) |
-			((guint32) bytes[6] << 16) |
-			((guint32) bytes[5] << 8) |
-			bytes[4];
-
-	if (length >= 10 &&
-		ifd_offset >= 6 &&
-		bytes[8] == 'C' &&
-		bytes[9] == 'R')
-		return "CR2";
```

Comment:
This was removed as it would be covered by:
https://github.com/ImageMagick/ImageMagick/blob/9b109732da21e1604ec28d32926c176bffb9d3e2/coders/dng.h#L20-L21